src/ImageStats/Histogram.h

```diff
@@ -18,25 +18,33 @@ struct HistogramResults {
     int num_bins;
     float bin_width;
     float bin_center;
-    std::vector<int> histogram_bins;
+    std::vector<int> histogram_bins; // histogram bin counts
 };
 
 class Histogram {
-    int _num_bins;
-    float _bin_width;
-    float _min_val;
-    std::vector<int> _hist;
+    float _min_val; // lower bound of the histogram (inclusive)
+    float _max_val; // upper bound of the histogram (inclusive)
+    HistogramResults _histogram; // histogram stats
     const std::vector<float>& _data;
 
+    bool ConsistencyCheck(const Histogram&, const Histogram&);
+
 public:
     Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data);
     Histogram(Histogram& h, tbb::split);
 
     void operator()(const tbb::blocked_range<size_t>& r);
-    void join(Histogram& h); // NOLINT
+    bool join(Histogram& h); // NOLINT
     void setup_bins();
 
-    HistogramResults GetHistogram() const;
+    HistogramResults GetHistogram() const { return _histogram; }
+    float GetMinVal() const { return _min_val; }
+    float GetMaxVal() const { return _max_val; }
+    int GetNbins() const { return _histogram.num_bins; }
```

Comment:
we should unify this, so that we don't need the `HistogramResults` struct. It doesn't really serve any purpose

Reply:
@veggiesaurus, if you and others agree, I could refactor more code. I would remove HistogramResults and keep the histogram object and its content. That would change several parts of the codebase so that instead of passing HistogramResults around, we can pass Histogram. With the right method calls, it should be even easier to read.

Reply:
> I would remove HistogramResults and keep the histogram object and its content. That would change several parts of the codebase so that instead of passing HistogramResults around, we can pass Histogram. With the right method calls, it should be even easier to read.

@jolopezl please do so. I think it's just a needless complication at this point.